### ISSUE TITLE
Why have a `print` function

### ISSUE THREAD

**Issue by Raynos** [ianb/doctestjs#36]
Why do

```
print(value)
// => 42
```

When you could just have 

```
// value => 42
```

**Comment by doboy** [comment]
+1

**Comment by ianb** [comment]
Print serves a lot of purposes, like in the Spy case.  But I can see how a second form could be interesting, in the general pattern of:

```
// (expression) => output
```

I worry about catching comments that weren't intended as expected output.  E.g,. if any `=>` appears anywhere in a comment a naive regex might catch it.  But I can imagine a limited regex that requires that `=>` appear on the same line as the expression (which can only be one line), and that gets translated to a print and expected output.

**Comment by Raynos** [comment]
@ianb 

> like in the Spy case.

this statement is vague and not obvious. You should explain it in more detail.

**Comment by ianb** [comment]
Spies work like this:

```
function Spy(name) {
  return function () {
    var a = repr(args);
    print(name + '(' + a.substr(0, a.length-2) + ')');
  };
}
```

That is, print how they are called but do not result in an expression, and the print statements are distinct and time-ordered.

**Comment by Raynos** [comment]
@ianb does this mean you have a special case for spies?

Or do you generically attach a `// =>` call below a function call where the last statement of the function is a print.

**Comment by ianb** [comment]
No, Spies actually call `print()`.  You might find the pseudocode in this section useful: http://doctestjs.org/reference.html#wait

Every block of code between `// =>` comments is executed in its entirety.  All calls the `print()` during that time is collected.  The text of all those calls is then compared to what is "expected" (i.e., the text after `// =>`).  Any code can call `print()`.  I think your project uses a very different way to do these comparisons.

**Comment by shamrin** [comment]
Random idea: we could magically add `print` if `=>` comment happens on the same line:

``` javascript
value // => 42
```

The above could be compiled to:

``` javascript
print(value)
// => 42
```

**Comment by ianb** [comment]
`value // => 42` would be cool, but not easy to implement.  But if someone wants to take it on, sure!

**Comment by shamrin** [comment]
@ianb, do you have any public doctest.js-using project? I would like to get a feeling if my `value // => 42` idea could be useful in practice. My use of doctest.js is too small right now :)

**Comment by ianb** [comment]
This is old-style (inline in HTML and using the doctest format), but it's longish: https://github.com/mozilla/receiptverifier/blob/master/test.html

I don't think this would be that helpful for that test – I watch for side effects a lot in it.

This one would be nice with this form: https://github.com/ianb/browsercommander/blob/master/static/tests/test_path.js

There's quite a few in here: https://github.com/ianb/thecutout/tree/master/cutout/tests/syncclient – mostly async so this form probably wouldn't be helpful.

**Comment by shamrin** [comment]
Here's test_path.js with the proposed change:

``` javascript
pathjoin('a', 'b/', 'c');      // => a/b/c
pathjoin('a', '/b/c', 'd');    // => /b/c/d
pathjoin('a/b/c', '../d');     // => a/b/d
pathjoin('a', '..');           // => a/..
pathjoin('/foo', '..', '..');  // => /
normpath('/a/b/..');           // => /a
normpath('/a/b/');             // => /a/b
normpath('/a/../../b/c/../d'); // => /b/d
normpath('/a/b///../c////');   // => /a/c
normpath('/a/././b/');         // => /a/b
```

Not bad, but I had to align outputs to make it readable.
